src/pages/CourseDetails/CourseDetails.tsx

```diff
@@ -0,0 +1,67 @@
+import React, {useEffect, useState} from 'react';
+import {useParams} from 'react-router-dom';
+import {getCourseDetailsById} from '../../services/courseService.ts';
```

Comment:
A few suggestions:

- setup src path re-writing through vite to avoid having to  reference your files by ../../.. 

-  import and re-export your components into the root of your components folder, that way you can import them all from the index file.  

- That requires you separating your universal components from your product components